### PR TITLE
Fixing Quota Retries for WriteBatch requests

### PR DESCRIPTION
**Pull request by phillipperalez**
* Making retry backoff configurable with a default of 3 seconds (old hardcode)
* Fixing Write Quota retries to include 500 with "Quota exceeded" as a valid error to retry
* Fix read retry to not DDOS on continual 429.

### CODE REVIEW

File: src/actions/google/drive/sheets/google_sheets.ts [hunk 1]
```diff
@@ -14,7 +14,9 @@ const MAX_REQUEST_BATCH = process.env.GOOGLE_SHEETS_WRITE_BATCH ? Number(process
 const MAX_ROW_BUFFER_INCREASE = 6000
 const SHEETS_MAX_CELL_LIMIT = 5000000
 const MAX_RETRY_COUNT = 5
+const RETRY_BASE_DELAY = process.env.GOOGLE_SHEETS_BASE_DELAY ? Number(process.env.GOOGLE_SHEETS_WRITE_BATCH) : 3
```
Comment (Jonting416) [inline src/actions/google/drive/sheets/google_sheets.ts:17]: Should this number be GOOGLE_SHEETS_BASE_DELAY as well

File: src/actions/google/drive/sheets/google_sheets.ts [hunk 2]
```diff
@@ -161,7 +163,7 @@ export class GoogleSheetsAction extends GoogleDriveAction {
         const spreadsheetId = files.data.files[0].id!
 
         const sheets = await this.retriableSpreadsheetGet(spreadsheetId, sheet,
-            0, request.webhookId!).catch((e: any) => {
+            1, request.webhookId!).catch((e: any) => {
```
Comment (Jonting416) [inline src/actions/google/drive/sheets/google_sheets.ts:166]: magic number - can we put this under a variable so we have context on what it means?  (I assume it's retry count here??)

Comment (Jonting416) [reply src/actions/google/drive/sheets/google_sheets.ts:166]: also are we just starting at retry 1 now?

Comment (phillipperalez) [reply src/actions/google/drive/sheets/google_sheets.ts:166]: It is retry count. This one is overkill to put into a const IMO

File: src/actions/google/drive/sheets/google_sheets.ts (hunk 1 above)
Comment (phillipperalez) [reply src/actions/google/drive/sheets/google_sheets.ts:17]: No. The env var is more specific in case I implement it for other actions.

Comment (jimbert) [reply src/actions/google/drive/sheets/google_sheets.ts:17]: Should this be its own envar then?

File: src/actions/google/drive/sheets/google_sheets.ts (hunk 2 above)
Comment (jimbert) [reply src/actions/google/drive/sheets/google_sheets.ts]: I think it's worth naming this for clarity

File: lib/actions/google/drive/sheets/google_sheets.js [hunk 3]
```diff
@@ -20,7 +20,9 @@ const MAX_REQUEST_BATCH = process.env.GOOGLE_SHEETS_WRITE_BATCH ? Number(process
 const MAX_ROW_BUFFER_INCREASE = 6000;
 const SHEETS_MAX_CELL_LIMIT = 5000000;
 const MAX_RETRY_COUNT = 5;
+const RETRY_BASE_DELAY = process.env.GOOGLE_SHEETS_BASE_DELAY ? Number(process.env.GOOGLE_SHEETS_WRITE_BATCH) : 3;
```
Comment (jimbert) [inline lib/actions/google/drive/sheets/google_sheets.js:23]: it could be helpful to include the time unit in the constant name

**Review by jimbert — DISMISSED** [review]
approved with minor nits

File: src/actions/google/drive/sheets/google_sheets.ts (hunk 1 above)
Comment (Jonting416) [reply src/actions/google/drive/sheets/google_sheets.ts:17]: I meant specifically on this line we reference two env vars, I figured they should be the same env var

**Review by Jonting416 — APPROVED** [review]

File: lib/actions/google/drive/sheets/google_sheets.js (hunk 3 above)
Comment (Jonting416) [reply lib/actions/google/drive/sheets/google_sheets.js]: +1 but can be done as a fast followup

**Pull request merged**